RoboJackets/robocup-software
# fix mypy errors in nmark_tactic.py

## Description
Basically fixed (or attempted to fix) the mypy static typing errors in the file tactic/nmark_tactic.py.  For specific details of changes see below and in Files changed.

## Associated Issue
#1741 foreach in all plays (tactic/nmark_tactic.py)

## Steps to test
Find "Pull request checks / mypy" and click details.

Expected result: Should run

## Sub-issues (Still open things that I think aren't actually issues)
Error: rj_gameplay/rj_gameplay/tactic/nmark_tactic.py:64: error: Module has no attribute "soccer" -> unsolved? Mypy error due to ros2 

Error: rj_gameplay/rj_gameplay/tactic/nmark_tactic.py:92: error: Argument 1 to "SkillEntry" has incompatible type "Mark"; expected "Type[Any]" -> In the skillEntry class that this is inherited from, it uses the TypeVar, which for some weird reason causes this incompatible type error because the variable type is generated at runtime making mypy get angry.
